src/DataTypes/Serializations/SerializationAggregateFunction.cpp

```diff
@@ -53,8 +54,21 @@ void SerializationAggregateFunction::deserializeBinary(IColumn & column, ReadBuf
     }
     catch (...)
     {
-        function->destroy(place);
-        throw;
+        // For approx_top_* states, optionally swallow malformed input and keep an empty state instead of failing CAST
+        const String & fname = function->getName();
+        const bool is_approx_top = fname.starts_with("approx_top_");
```

Comment:
This code looks like a bomb waiting to explode. We should never ignore malformed input